jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -240,6 +251,42 @@ def vue_data_selected(self, event):
 
         self._spectrum1d = selected_spec
 
+        # Also set the spectral min and max to default to the full range
+        cube = self._spectrum1d.get_object(cls=SpectralCube)
+        self.selected_subset = "None"
+        self._window = None
+        self.spectral_min = cube.spectral_axis[0].value
+        self.spectral_max = cube.spectral_axis[-1].value
+        self.spectral_unit = str(cube.spectral_axis.unit)
```

Comment:
```suggestion
        self.selected_subset = "None"
        self._window = None
        self.spectral_min = selected_spec.spectral_axis[0].value
        self.spectral_max = selected_spec.spectral_axis[-1].value
        self.spectral_unit = str(selected_spec.spectral_axis.unit)
```

No need to get the whole cube, we can use the collapsed Spectrum1D to get the spectral bounds (this fails as written currently).

Reply:
> this fails as written currently

I was gonna apply your suggestion but now I am not sure... LoL.

Reply:
Oh sorry, I meant your code fails and this suggestion fixes it.

Reply:
Ah, gotcha. English can be so confusing to me sometimes! 😸